AsyncDisplayKit/ASMultiplexImageNode.mm

```diff
@@ -293,9 +294,8 @@ - (void)displayWillStart
 - (void)visibilityDidChange:(BOOL)isVisible
 {
   [super visibilityDidChange:isVisible];
-  
+  ASDN::MutexLocker l(_downloadIdentifierLock);
```

Comment:
Does this need to be moved up? Is the idea to make sure visibilityDidChange occurs in its entirety? Otherwise this just means the lock is held longer since it's also locked in _updateProgressImageBlockOnDownloaderIfNeeded.